cpp/src/io/orc/writer_impl.cu

```diff
@@ -1077,6 +1077,9 @@ void writer::impl::write(table_view const &table)
 
   auto streams  = create_streams(orc_columns, stripe_bounds);
   auto enc_data = encode_columns(*device_columns, orc_columns, str_col_ids, stripe_bounds, streams);
+  dict_data.release();
+  dict_index.release();
+  stream.synchronize();
```

Comment:
Oh, I completely misunderstood the last commit. The idea was to release before the sync IN the encode_columns.
```
  gpu::EncodeOrcColumnData(chunks, chunk_streams, stream);
  dict_data.release();
  dict_index.release();
  stream.synchronize();
  ```
  You should still move dict_data and dict_index into encode_columns.